drivers/hmis/app/jobs/hmis/assessment_questions_job.rb

```diff
@@ -4,100 +4,89 @@
+          ce_assessment.assessment_questions.delete_all
+
+          questions_to_import = [].tap do |questions|
+            question_items = definition.link_id_item_hash.values.select { |item| item.mapping&.custom_field_key.present? }
+            question_items.each_with_index do |item, index|
+              next unless item.mapping&.custom_field_key
+
+              key = item.mapping.custom_field_key
+              cded = custom_data_element_definitions_by_key[key]
+              raise "No custom data element definition found for key: #{key}" unless cded
 
-            questions << assessment.assessment_questions.build(
-              enrollment_id: assessment.enrollment_id,
-              personal_id: assessment.personal_id,
-              user_id: assessment.user_id,
-              date_created: assessment.assessment_date,
-              date_updated: Time.current,
+              # find responses to this question
+              value = custom_assessment.custom_data_elements.select { |cde| cde.data_element_definition_id == cded.id }&.map(&:value)
+              value = nil if value.blank? # [] => nil
+              value = value.first if value&.size == 1 # [value] => value
+              value = yes_no_nil(value) if cded.field_type.to_sym == :boolean # false => 'No'
 
-              assessment_question: key,
-              # Truncate to ensure value is not too long for db
-              assessment_answer: value&.to_s&.truncate(500),
-              assessment_question_group: question_group(key),
-              assessment_question_order: question_order(key),
-            )
+              questions << ce_assessment.assessment_questions.build(
+                enrollment_id: ce_assessment.enrollment_id,
+                personal_id: ce_assessment.personal_id,
+                user_id: ce_assessment.user_id,
+                date_created: ce_assessment.date_created,
+                date_updated: ce_assessment.date_updated,
+
+                # Note: this key is referenced by the tc_hat calculator
+                assessment_question: key,
+                # Truncate to ensure value is not too long for db
+                assessment_answer: value&.to_s&.truncate(500),
+                # Name of the section that this question belongs to
+                assessment_question_group: definition.link_id_section_hash[item.link_id],
+                # Order of this question in the form
+                assessment_question_order: index + 1,
+              )
+            end
           end
+          ::GrdaWarehouse::Hud::AssessmentQuestion.import!(questions_to_import)
         end
-        ::GrdaWarehouse::Hud::AssessmentQuestion.import!(questions_to_import)
       end
     end
 
-    private def question_type(key)
-      form_definition[key].try(:[], :type)
+    private def custom_assessment_scope
+      @custom_assessment_scope ||= Hmis::Hud::CustomAssessment.where(id: @custom_assessment_ids).
```

Comment:
Probably don't need to memoize the scope since it shouldn't be instantiated at this point.